examples/allocation/tinygo/testdata/greet.go

```diff
@@ -77,3 +80,19 @@ func stringToPtr(s string) (uint32, uint32) {
 	unsafePtr := uintptr(unsafe.Pointer(ptr))
 	return uint32(unsafePtr), uint32(len(buf))
 }
+
+// stringToNativePtr returns a pointer and size pair for the given string in a way
+// compatible with WebAssembly numeric types. The pointer is not automatically
+// managed by TinyGo hence it must be freed by the host.
+func stringToNativePtr(s string) (uint32, uint32) {
+	if len(s) == 0 {
+		return 0, 0
```

Comment:
Is this special case valid? if so add a trailing comment why (for example,`// CGO.free ignores a pointer of zero`)
If not, delete the special case.

Reply:
As far as I recall, calling `malloc` with a size equals to 0, is implementation dependent hence I opted not to call it in such case. 

I Will add a note if that's fine with you but in this case it is probably never happening.

Reply:
This is an example, so we can't really add suggestions to special case zero unless it will work. We don't have to consider externally provided impls of malloc/free in the example, though. What tinygo uses with this example and flags given to build is enough.

@anuraaga @dgryski do either of you happen to know what's better:

* not dodging calling malloc dependent on length
* dodging calling malloc with zero length (results in free called with zero)

Reply:
here's the tinygo code in question. Personally, I prefer to not special case without saying why we are doing it. Most people won't know why they are calling malloc/free in the first place and code like this gets copy/pasted around. As best I can tell, we are assuming the result will be zero and so we are returning that instead of calling things. That would be a perf optimization, which we'd comment why.

If we don't have a good reason tied to tinygo, I prefer we don't do any special casing in this function. Without comments, special casing raises more questions than answers, especially that a zero pointer is something safe to return back to tinygo on free. For example, pointers are often memory offsets in wasm, so zero would be the initial memory position. While free shouldn't have any side effects there and it is a safe guess maybe to assume it will do nothing on free.. it is a lot better to leave the problem on what to do with length zero completely up to tinygo (e.g. don't special case). Cheaper to review, less code, easier example.

https://github.com/tinygo-org/tinygo/blob/0d56dee00f49bd50eb373c02c30062a75ec28f10/src/runtime/arch_tinygowasm_malloc.go#L13-L34

Reply:
remove d the special case